File: 0_100/20.js
/**
 * @param {string} s
 * @return {boolean}
 */
 var isValid = function(s) {
    let stack = [];

    for(let i = 0; i < s.length;i++) {
        if(s[i] === '(' || s[i] === '{' || s[i] === '[') {
            stack.push(s[i]);
        } else if(s[i] === ')' && stack.pop() !== '(') {
            return false;
        } else if(s[i] === '}' && stack.pop() !== '{') {
            return false;
        } else if(s[i] === ']' && stack.pop() !== '[') {
            return false;
        }
    }

    if(stack.length > 0) {
        return false;
    }
    return true;
};

console.log(isValid("("));

File: 1_100/20.cs
public class Solution {
    public bool IsValid(string s) {
        if(s.Length == 1) {
            return false;
        }
        
        Stack<char> openParenStack = new Stack<char>();
        Dictionary<char, char> parens = new Dictionary<char, char>();
        
        parens.Add(')','(');
        parens.Add(']','[');
        parens.Add('}','{');
        
        for(int i = 0; i < s.Length;i++) {
            if(s[i] == '(' ||s[i] == '{' || s[i] == '[') {
                openParenStack.Push(s[i]);
            } else if(s[i] == ')' || s[i] == '}' || s[i] == ']') {
                if(openParenStack.Count == 0) {
                    return false;
                }
                var paren = openParenStack.Pop();
                if(paren != parens[s[i]]) {
                    return false;
                }
            }
        }
                   
        return openParenStack.Count == 0;
    }
}